compliance/collection/nodeinventorizer/nodeinventory.go

```diff
@@ -58,23 +58,42 @@ func protoComponentsFromScanComponents(c *nodes.Components) *scannerV1.Component
 	return protoComponents
 }
 
-func convertRHELComponents(rc *database.RHELv2Components) []*scannerV1.RHELComponent {
+func convertAndDedupRHELComponents(rc *database.RHELv2Components) []*scannerV1.RHELComponent {
 	if rc == nil || rc.Packages == nil {
 		log.Warn("No RHEL packages found in scan result")
 		return nil
 	}
-	v1rhelc := make([]*scannerV1.RHELComponent, 0, len(rc.Packages))
-	for _, rhelc := range rc.Packages {
-		v1rhelc = append(v1rhelc, &scannerV1.RHELComponent{
-			// TODO(ROX-13936): Find out if ID field is needed here
+
+	convertedComponents := make(map[string]*scannerV1.RHELComponent, 0)
+	for i, rhelc := range rc.Packages {
+		comp := &scannerV1.RHELComponent{
+			// The loop index is used as ID, as this field only needs to be unique for each NodeInventory result slice
+			Id:          int64(i),
 			Name:        rhelc.Name,
 			Namespace:   rc.Dist,
 			Version:     rhelc.Version,
 			Arch:        rhelc.Arch,
 			Module:      rhelc.Module,
 			Cpes:        rc.CPEs,
 			Executables: rhelc.Executables,
-		})
+		}
+		compKey := makeComponentKey(comp)
+		if compKey != "" {
+			if _, contains := convertedComponents[compKey]; !contains {
+				log.Debugf("Adding component %v to convertedComponents", comp.Name)
+				convertedComponents[compKey] = comp
+			} else {
+				log.Warnf("Detected package collision in Node Inventory scan. Skipping package %s at index %d", compKey, i)
+			}
+		}
+
+	}
+	return maps.Values(convertedComponents)
+}
+
+func makeComponentKey(component *scannerV1.RHELComponent) string {
```

Comment:
nit: Maybe make this return a struct instead of a string

Reply:
I could make it return a struct, but then I'd either need to convert it to a string in the caller, or switch the callers' `convertedComponents` map to take a struct as key.
The latter would bring us back to the idea discussed [here](https://github.com/stackrox/stackrox/pull/4286#discussion_r1065254337), which is a viable alternative.

But doing that introduces a possible problem:
We're considering the tuple (name,version,arch,module) for uniqueness checks. `RHELComponents` does have more fields than these four, which could lead to us producing duplicates, e.g. when attached `Executables`differ. To the dedup code here it would be two separate entries, whereas the code that uses the tuple would see it as a collision.

Would you be okay with redesigning this when we move the conversion functions into Scanner?